### PR TITLE
AGENT-966: Reduce dependency from kube-system/cluster-config-v1 when generating the image 

### PR DESCRIPTION
**Pull request by andfasano**
The following patch refactored the logic in the `ClusterInfo` asset for retrieving information from the `kube-system/cluster-config-v1`, for the following entities:

- ImageDigestMirrorSets
- ImageContentPolicies
- APIDNSName
- ClusterName
- SSH key
- FIPS

Thus removing completely the dependency on the `kube-system/cluster-config-v1` resource.

*Tip:* See individual commits for the specific details related to each field

### CODE REVIEW

**Comment by andfasano** [comment]
/hold

Requires rbac update first in the oc command for passing the add nodes CI job

**Comment by andfasano** [comment]
Requires https://github.com/openshift/oc/pull/1895

**Comment by andfasano** [comment]
cc @zaneb

File: pkg/asset/agent/joiner/clusterinfo.go [hunk 1]
```diff
@@ -295,8 +299,20 @@ func (ci *ClusterInfo) retrieveInstallConfigData() error {
 	if err = yaml.Unmarshal([]byte(data), &installConfig); err != nil {
 		return err
 	}
+	return nil
+}
+
+func (ci *ClusterInfo) retrieveFIPS() error {
+	workerMachineConfig, err := ci.OpenshiftMachineConfigClient.MachineconfigurationV1().MachineConfigs().Get(context.Background(), "99-worker-fips", metav1.GetOptions{})
+	if err != nil {
+		// This resource is not created when FIPS it's not enabled
```
Comment (rwsu) [inline pkg/asset/agent/joiner/clusterinfo.go:244]: s/it's/is?

**Review by rwsu — COMMENTED** [review]
Small nit.

/approve

**Pull request merged**